app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala

```diff
@@ -22,20 +33,60 @@ class OAuth2TokenValidationService(oAuth2ProviderSettings: OAuth2ProviderSetting
     validationResult.map(getUserId)
   }
 
-  def validateTokenWithJWT(accessToken : String) = {
-     Future.failed(new Throwable);
+  def validateTokenWithJWT(accessToken: String): Future[Either[Exception, ValidationResponse]] = {
+      processJWT(accessToken, publicKey)
+  }
+
+  def fetchPublicKey(url: String): Future[RSAPublicKey] = {
+    WS.url(url)
+      .getStream()
+      .flatMap(f => generatePublicKeyFromBytes(f))
+  }
+
+  def generatePublicKeyFromBytes(f: (WSResponseHeaders, Enumerator[Array[Byte]])): Future[RSAPublicKey] = {
+    val (headers, responce) = f
+
+    val consume = Iteratee.consume[Array[Byte]]()
+    responce(consume)
+      .flatMap(i => i.run)
+      .map(key => {
+      val kf = KeyFactory.getInstance("RSA")
+      kf.generatePublic(new X509EncodedKeySpec(key)).asInstanceOf[RSAPublicKey]
+    })
+  }
+
+  def processJWT(accessToken: String, key: RSAPublicKey): Future[Either[Exception, ValidationResponse]] = {
+    Try(JWSObject.parse(accessToken))
+      .map(jwtObject => verifyToken(jwtObject, key)) match {
+      case Success(e) => Future.successful(e)
+      case Failure(ex) => Future.failed(TokenParseException(ex.getMessage))
```

Comment:
here you return a Future.failed. Why do you still have the Future[Either[..., ...]] in your method signature? Can you try simplifying the types to always use one or the other approach? Or at least a sensible one like
failed Future == unexpected exception
Left(Exception) == expected (e.g. TokenParseException) exception.